tez-dag/src/main/java/org/apache/tez/client/LocalClient.java

```diff
@@ -438,7 +447,9 @@ public DAGStatus apply(Set<StatusGetOpts> statusOpts, Long timeout) {
   public boolean shutdownSession(Configuration configuration, ApplicationId sessionAppId,
       UserGroupInformation ugi) throws TezException, IOException, ServiceException {
     if (isLocalWithoutNetwork) {
-      clientHandler.shutdownAM();
+      if (clientHandler != null){
+        clientHandler.shutdownAM();
+      }
       return true;
     } else {
```

Comment:
If - else can be simplified here (else not needed)

Reply:
right, updating it